### DISCUSSION TITLE
Favorites, Routes, Tracks - visible depending on zoom

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#335] [Q&A] [answered]
Now that we have the ability to create collections of our places, maybe we could show them on a map depending on zoom level? As is the case with VIA points and shaping points along the route.

We currently have two states: visible/invisible. This could be replaced with a zoom slider, at the extremes of which it would be always/never. I think that an independent setting could apply to individual groups (Favorites) or folders (Tracks, Routes).

**Comment by devemux86** [comment]
@Ciutnik thanks for the suggestion.

This seems useful, especially when displaying too many favorites (without clustering) or very large tracks
at small zoom levels it affects map performance.

Note: favorites visibility is set by group, routes / tracks visibility is set by route / track.

**Reply by Ciutnik** [reply]
> One route is active at a time

I thought about it before I submitted the request. Initially, I wanted to suggest only Favorites and Tracks, but maybe some visualization of the Route (without activating it) would be possible? But this is rather a big complication, I guess a "preview" or something like that would have to be stored for each Route... For Favorites and Tracks, it's easier to imagine 🙂

**Reply by devemux86** [reply]
> maybe some visualization of the Route (without activating it) would be possible?

(also mentioned here: https://github.com/devemux86/cruiser/discussions/35#discussioncomment-7088902)

We can already use the "Show on map" on the favorite routes with options:
- **Route**: offline and ready for navigation
- **Track**: overlay their path with any color (visible while the app runs)

**Reply by Ciutnik** [reply]
> Track: overlay their path with any color (visible while the app runs)

Exactly! And this is what the visibility option depending on the zoom could be about 🙂

**Reply by devemux86** [reply]
For route overlays, please see here:
- https://github.com/devemux86/cruiser/discussions/35#discussioncomment-7166440

**Comment by devemux86** [comment]
Visibility of favorites + routes + tracks can be adjusted by selecting the zoom level:

<img src="https://github.com/devemux86/cruiser/assets/3484020/abdd1fb5-9bb7-46b4-9485-d250ee59260c" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/e1b97337-5342-4cb9-ac9a-ea3f1a01465b" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/48f7bb19-71f5-402e-ab5e-b099e05d1355" width="33%"/>

**Reply by Ciutnik** [reply]
Cool! It looks like the zoom level will be set for each element/group separately 👍

**Reply by Ciutnik** [reply]
Considering users who like simplicity, I wonder if there should be an "Apply to all" button (or checkbox). This may be important if you have a large collection. Setting individual zoom levels can be frustrating for some.

Maybe instead of the "Cancel" button, add a back arrow, which is already used in other windows, and replace it with this button?

By the way, "Apply to all" would be nice to have in the Profiles settings 🙂

**Reply by devemux86** [reply]
@Ciutnik do you mean the folders with routes or tracks?

I thought that too. Do other apps have such a global action for the entire favorite folders?
However, displaying an entire folder with too many routes or tracks could hurt performance.
(we should be careful with making too many layers visible)

The back arrow returns to the parent screen, the cancel closes the screen,
or do you mean something else?

Also what do you mean about the profiles?

**Reply by Ciutnik** [reply]
> do you mean the folders with routes or tracks?

To maintain consistency - for favorites, routes and tracks 🙂 I was thinking about the possibility of setting the same zoom for all elements in a group/folder in one click. However, I would like to point out that I personally prefer the ability to set the zoom for each element separately (especially for Favorites folders).

> Also what do you mean about the profiles?

I'm thinking about the possibility of applying some setting to all Profiles, in one click. We have a similar solution in OsmAnd. E.g. If I wanted to change the sizes of icons, buttons or fonts in all Profiles. Now we need to change it in each Profile one by one, but it would be more convenient to check option "Apply to all". Of course, this is only for convenience, and only sometimes, so it doesn't really matter 🙂

**Reply by devemux86** [reply]
Note that favorites vs routes / tracks are set up a bit differently:
- **Favorites**: the group defines the icon / color / visible

- **Routes / Tracks**: the units define the color / visible

**Reply by Ciutnik** [reply]
Yes of course. These are different settings. I mean that if you want to have, for example, the same "Display size" setting in each Profile – you set the desired value in one profile, and select the option "apply to all profiles". Simple saving of clicks, but this is only a minor improvement, but maybe an unnecessary complication.

**Reply by devemux86** [reply]
Regarding profiles, this would mean that changes to any setting would have an additional question if it would apply to all profiles.
We could see it, how it is possible in the future. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
> additional question if it would apply to all profiles

A smart and good solution! Wouldn't complicate the UI in any way! 👍

**Reply by devemux86** [reply]
It could complicate things, that's why I have to think all the options. :slightly_smiling_face:
Also it can annoy users, as the setting changes will not be applied immediately,
but a question will happen every time on each setting change.

**Comment by DolomitesTom** [comment]
Would it be possible to update a changed route from the favorites? I mean show on map , change something, update again in the favorites.

**Reply by devemux86** [reply]
Edit / update of favorite routes will be available in the next update. I describe the process here:
- https://github.com/devemux86/cruiser/discussions/336#discussioncomment-7223205

**Comment by devemux86** [comment]
As discussed, we can improve the route + track group screens.

- Show how many group items are visible
- Change the color / visibility of the entire group

<img src="https://github.com/devemux86/cruiser/assets/3484020/dcec9b30-926e-4d6c-aa2f-24aea5031a28" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/c0d96ed4-495d-4572-8760-ba973d0b9ff6" width="50%"/>

**Comment by devemux86** [comment, marked as answer]
> Favorites, Routes, Tracks - visible depending on zoom

Available in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
